### PR TITLE
insertion_sort_recursive

### PR DESCRIPTION
**Pull request by itsvinayak**
#### Description of Change

Adding algo for insertion sort using recursive

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C/pull/647"></a>

### CODE REVIEW

**Comment by itsvinayak** [comment]
#633

File: sorting/insertion_sort_recursive.c [hunk 1]
```diff
@@ -0,0 +1,72 @@
+
```
Comment (kvedala) [inline sorting/insertion_sort_recursive.c:1]:
```suggestion
```

File: sorting/insertion_sort_recursive.c [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+static void test()
+{
+    const int size = rand() % 500; /* random array size */
+    int *arr = (int *)calloc(size, sizeof(int));
+
+    /* generate size random numbers from -50 to 49 */  
+    for (int i = 0; i < size; i++)
+    {
+        arr[i] = (rand() % 100) - 50;/* signed random numbers */
+        // convert -ve no. to positive
```
Comment (kvedala) [inline sorting/insertion_sort_recursive.c:51]: why convert -ve numbers? The insertion sort should work for negative numbers as well, right?

File: sorting/insertion_sort_recursive.c [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        }
+    }
+    RecursionInsertionSort(arr, size);
+    for (int i = 0; i < size  ; ++i)
+    {
+        assert(arr[i] <= arr[i + 1]);
+    }
+    free(arr);
+}
+/** Main function
```
Comment (kvedala) [inline sorting/insertion_sort_recursive.c:63]:
```suggestion

/** Main function
```

**Review by kvedala — CHANGES_REQUESTED** [review]
👍

**Comment by itsvinayak** [comment]
hey @kvedala, I have made the necessary changes
 please have a look

thanks
@itsvinayak

File: sorting/insertion_sort_recursive.c (hunk 2 above)
Comment (itsvinayak) [reply sorting/insertion_sort_recursive.c]: @kvedala changes made

File: sorting/insertion_sort_recursive.c [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+ * @file
+ * @brief [Insertion sort](https://en.wikipedia.org/wiki/Insertion_sort)
+ * algorithm implementation.
+ */
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+/**
```
Comment (kvedala) [inline sorting/insertion_sort_recursive.c:11]:
```suggestion
/**
 * @addtogroup sorting Sorting algorithms
 * @{
 */
/**
```

File: sorting/insertion_sort_recursive.c [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    int j = size-2;
+    // swapping logic for insertion sort
+    while(j >= 0 && arr[j] > key)
+    {
+        arr[j+1] = arr[j];
+        j--;
+    }
+    arr[j+1] = key;
+}
+
```
Comment (kvedala) [inline sorting/insertion_sort_recursive.c:37]:
```suggestion
/** @} */
```

File: sorting/insertion_sort_recursive.c [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+{
+    if(size <= 0)
+    {
+        return;
+    }
+    
+    // marking recursive call to reach starting element
+    RecursionInsertionSort(arr,size-1);
+    
+
```
Comment (kvedala) [inline sorting/insertion_sort_recursive.c:26]:
```suggestion
```

**Review by kvedala — CHANGES_REQUESTED** [review]
👍 
last fix: grouping the function helps in generating more structured documentation.

**Review by kvedala — APPROVED** [review]

**Pull request merged**